processor/src/main/java/dz/jtsgen/processor/renderer/module/tsd/TSDGenerator.java

```diff
@@ -105,7 +105,7 @@ private void outputTypes(TSModuleInfo module, int ident, List<TSType> types, Pri
     }
 
     private TSTypeVisitor creatTypeVisitor(TSModuleInfo module, PrintWriter out) {
-        return module.generateTypeGuards() ? new TSTypeVisitorTypeGuards(out) : new TSTypeVisitorDefault(out);
+        return module.generateTypeGuards() ? new TSTypeVisitorTypeGuards(out, model) : new TSTypeVisitorDefault(out, model);
```

Comment:
I think, the dependency to model should be removed. I'll accept that at first, but it'l removed afterwards.

Reply:
That makes sense. Perhaps we should extend the TSEnumMember to provide it? So we do not have these kind of dependency.